newIDE/app/src/EventsFunctionsExtensionEditor/index.js

```diff
@@ -602,6 +602,22 @@ export default class EventsFunctionsExtensionEditor extends React.Component<
     done(true);
   };
 
+  _onEventsBasedBehaviorPasted = (
+    eventsBasedBehavior: gdEventsBasedBehavior,
+    sourceExtensionName: string
+  ) => {
+    const { project, eventsFunctionsExtension } = this.props;
+    if (eventsFunctionsExtension.getName() === sourceExtensionName) {
+      return;
+    }
+    gd.WholeProjectRefactorer.updateExtensionNameInEventsBasedBehavior(
+      project,
+      eventsFunctionsExtension,
+      eventsBasedBehavior,
+      sourceExtensionName
+    );
+  };
```

Comment:
It should probably do a refresh of the extension metadata here because otherwise most instruction in the pasted behavior will appear as unknown until the tab is switched.